src/main/java/org/cryptomator/ui/mainwindow/VaultDetailUnlockedController.java

```diff
@@ -54,6 +98,33 @@ public void showVaultStatistics() {
 		vaultStats.getUnchecked(vault.get()).showVaultStatisticsWindow();
 	}
 
+	private boolean containsUnlockedVaultContent(Path path) {
+		return path.startsWith(vault.get().getAccessPoint());
+	}
+
+	private Optional<Path> getCiphertextPath(Path path) {
+		if (!containsUnlockedVaultContent(path)) {
+			LOG.debug("Path does not start with access point of selected vault: {}", path);
+			return Optional.empty();
+		}
+		try {
+			var cleartextPathString = path.toString().substring(vault.get().getAccessPoint().length());
+			if (!cleartextPathString.startsWith("/")) {
+				cleartextPathString = "/" + cleartextPathString;
+			}
+			var cleartextPath = Path.of(cleartextPathString);
+			return Optional.of(vault.get().getCiphertextPath(cleartextPath));
+		} catch (IOException e) {
+			LOG.debug("Unable to get ciphertext path from path: {}", path);
+			return Optional.empty();
+		}
+	}
+
+	private void revealPath(Path path) {
+		// Unable to use `application.get().getHostServices().showDocument()` here, because it does not reveal files, only folders.
+		Desktop.getDesktop().browseFileDirectory(path.toFile());
```

Comment:
`Desktop::browseDirectory` is not supported on Windows.

Reply:
And also not on Ubuntu

Reply:
An Alternative for linux would involve d-bus: https://freedesktop.org/wiki/Specifications/file-manager-interface/

Usage example: 
```
dbus-send --print-reply --dest=org.freedesktop.FileManager1 /org/freedesktop/FileManager1 org.freedesktop.FileManager1.ShowItems array:string:"file:///home/john.doe/foo.txt" string:"
```

(from https://askubuntu.com/questions/133597/reveal-file-in-file-explorer)

Reply:
AWT call is replaced with RevealPathService in https://github.com/cryptomator/cryptomator/pull/2592/commits/c4cf51adc1df9aea7251da60917e96f6023537a4